ta/avb/entry.c

```diff
@@ -19,28 +23,78 @@ static TEE_Result get_slot_offset(size_t slot, size_t *offset)
 	return TEE_SUCCESS;
 }
 
-static TEE_Result create_state(uint32_t lock_state, TEE_ObjectHandle *h)
+static TEE_Result create_rb_state(uint32_t lock_state, TEE_ObjectHandle *h)
 {
 	const uint32_t flags = TEE_DATA_FLAG_ACCESS_READ |
 			       TEE_DATA_FLAG_ACCESS_WRITE |
 			       TEE_DATA_FLAG_OVERWRITE;
 
-	return TEE_CreatePersistentObject(storageid, obj_name, sizeof(obj_name),
-					  flags, NULL, &lock_state,
-					  sizeof(lock_state), h);
+	return TEE_CreatePersistentObject(storageid, rb_obj_name,
+					  sizeof(rb_obj_name), flags, NULL,
+					  &lock_state, sizeof(lock_state), h);
 }
 
-static TEE_Result open_state(uint32_t default_lock_state, TEE_ObjectHandle *h)
+static TEE_Result open_rb_state(uint32_t default_lock_state,
+				TEE_ObjectHandle *h)
 {
-	uint32_t flags = TEE_DATA_FLAG_ACCESS_READ | TEE_DATA_FLAG_ACCESS_WRITE;
+	uint32_t flags = TEE_DATA_FLAG_ACCESS_READ |
+			 TEE_DATA_FLAG_ACCESS_WRITE;
 	TEE_Result res;
 
-	res = TEE_OpenPersistentObject(storageid, obj_name,
-				       sizeof(obj_name), flags, h);
+	res = TEE_OpenPersistentObject(storageid, rb_obj_name,
+				       sizeof(rb_obj_name), flags, h);
+	if (!res)
+		return TEE_SUCCESS;
+
+	return create_rb_state(default_lock_state, h);
+}
+
+static TEE_Result create_named_value(char *obj_name, char *value,
+				     size_t value_size, TEE_ObjectHandle *h)
+{
+	const uint32_t flags = TEE_DATA_FLAG_ACCESS_READ |
+			       TEE_DATA_FLAG_ACCESS_WRITE |
+			       TEE_DATA_FLAG_OVERWRITE;
+
+	if (strlen(obj_name) >= TEE_OBJECT_ID_MAX_LEN)
+		return TEE_ERROR_ITEM_NOT_FOUND;
+
+	return TEE_CreatePersistentObject(storageid, obj_name,
+					  strlen(obj_name) + 1,
+					  flags, NULL, value,
+					  value_size, h);
+}
+
+static TEE_Result open_named_value(char *obj_name, uint32_t obj_max_size,
+				   char *value, size_t value_size,
```

Comment:
Since `value` will only be used if `create` is `true`, perhaps `value` could be used instead to indicate if an object should be created with a default value or not?

Reply:
Frankly, I don't see any case when it can be used.
If the `write_persistent_storage()` AVB cmd is called with `value == NULL` or `value_size == 0`, we have to return `AVB_IO_RESULT_ERROR_NO_SUCH_VALUE` and that's it and we shouldn't create any objects with `default` values.
https://github.com/u-boot/u-boot/blob/master/lib/libavb/avb_ops.h#L266

Reply:
re-implemented as you described